### PR TITLE
PDI-18 changes standard and changes negotiated price

### PR DESCRIPTION
**Pull request by CarlHA**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 1]
```diff
@@ -75,128 +78,162 @@ public string LearnerReferenceNumberStub()
...
+            }
+            else if (mainAims.First().StdCodeSpecified)
+            {
+                componentAims.ForEach(ca => ca.LearnAimRef = "00300545");
+            }
+        }
+
+        private void CorrectComponentAimStartAndEndDates(MessageLearner messageLearner)
+        {
+            var mainAims = messageLearner.LearningDelivery.ToList().Where(aim => aim.AimType == 1);
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Move `ToList` to the end of line

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 2]
```diff
@@ -75,128 +78,162 @@ public string LearnerReferenceNumberStub()
...
+                tnp.AFinDateSpecified = true;
+            }
         }
 
-        protected void RemovePmrRecord(MessageLearner learner)
+        private void SetComponentAimLearnAimReference(MessageLearner messageLearner)
         {
-            var deliveries = learner.LearningDelivery.ToList().Where(ld => ld.AimType == 1);
-            deliveries.ForEach(d=>d.AppFinRecord = d.AppFinRecord.ToList().Where(af => af.AFinType != LearnDelAppFinType.PMR.ToString()).ToArray());
+            var mainAims = messageLearner.LearningDelivery.ToList().Where(aim => aim.AimType == 1);
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: moved

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs (hunk 1 above)
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: moved

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs [hunk 3]
```diff
@@ -75,128 +78,162 @@ public string LearnerReferenceNumberStub()
...
+            var componentAims = messageLearner.LearningDelivery.ToList().Where(aim => aim.AimType == 3);
+
+            // case for 205 only - this logic needs to be improved!
+            if (mainAims.Count() == 2 && componentAims.Count() == 1 && (mainAims.Any(a=>a.StdCode == 52) || mainAims.Any(a=>a.StdCode == 53)))
```
[the remark this answers is not in the document]
Comment (CarlHA) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/LearnerMutators/FM36Base.cs]: Frameworks and Standards operate differently with regard to LearnAimRef for the component aim.  Frameworks need specific codes, however this is not the case for standards. This conditional logic is not required and StandardLearner has been modified accordingly.

**Comment by CarlHA** [comment]
Replacd by #457

**Pull request closed without merging**